### PR TITLE
mail-mta/notqmail: new package

### PR DESCRIPTION
**Pull request by DerDakon**
["notqmail is a community-driven fork of qmail, beginning where netqmail left off".](https://notqmail.org)

From everything we can tell there will be no new release of netqmail, so we decided to do our own thing. Eventually this will replace the existing mail-mta/netqmail ebuilds.

### CODE REVIEW

**Comment by DerDakon** [comment]
Can anyone tell me how virtual/qmail/qmail-1.03.ebuild needs to be updated for this?

**Comment by juippis** [comment]
I **think** virtuals are excluded from the equal visibility requirement. If that's true, you can just add notqmail there. What I find odd is that it has hardcoded version dependencies though. You should also update the EAPI while at it. 

If CI bot gives an error, its then most likely due to stable or missing arch KEYWORDS. Then, make a revbump or even version bump to the virtual while resetting keywords. Hope it doesnt come to profile masking arches. But really, it should just work.

**Comment by DerDakon** [comment]
This now mixes in another commit, just to see what the CI bot says. Will raise it for discussion on the ML afterwards.

**Comment by mgorny** [comment]
> I **think** virtuals are excluded from the equal visibility requirement. If that's true, you can just add notqmail there. What I find odd is that it has hardcoded version dependencies though. You should also update the EAPI while at it.

Any packages with `||` deps need only one of the deps to have matching keyword, for each keyword in your package. The difference with virtuals is that generally you can update keywords in virtual to much union of keywords of all providers, without stablereqs and all that.

**Comment by DerDakon** [comment]
Depends on #13551 for proper merging (change is included here, but this PR should focus only on the last 2 commits).

**Comment by juippis** [comment]
I just noticed something. We don't add -9999 version ONLY new packages, you need to take some sort of snapshot and package that if upstream hasn't made any release. 

Then I'd ask you to rename that "big-todo.103.patch" to something more recognizable and related to qmail, just appending ${PN} in front of it in distdir is enough.

**Comment by juippis** [comment]
I just noticed something. We don't add -9999 version ONLY new packages, you need to take some sort of snapshot and package that if upstream hasn't made any release. 

Then I'd ask you to rename that "big-todo.103.patch" to something more recognizable and related to qmail, just appending ${PN} in front of it in distdir is enough.

**Comment by DerDakon** [comment]
That patch is just copied from netqmail, where it has the same name forever.

I just wait for us to do a 1.08 as there are already several fixes in the master branch that I don't want to backport. Delaying this until 1.08 is out is fine with me, if I can reasonably expect a quick merge then.

**Comment by juippis** [comment]
Sounds good to me. Ping me when you push and I see it faster (should respond within 24 hours).

File: mail-mta/notqmail/notqmail-9999.ebuild [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+GENQMAIL_PV=20191010
+QMAIL_SPP_PV=0.42
+
+QMAIL_TLS_PV=20190517
+QMAIL_TLS_F=notqmail-1.07-tls-${QMAIL_TLS_PV}.patch
+
+QMAIL_BIGTODO_PV=103
+QMAIL_BIGTODO_F=big-todo.${QMAIL_BIGTODO_PV}.patch
+
+QMAIL_LARGE_DNS='qmail-103.patch'
```
Comment (juippis) [inline mail-mta/notqmail/notqmail-9999.ebuild]: Is there a reason not using `" "` here? (only eyeable small nitpick I found)

Comment (DerDakon) [reply mail-mta/notqmail/notqmail-9999.ebuild]: just copied from netqmail, will change that to ""

**Comment by DerDakon** [comment]
@juippis ready for take off.

**Comment by DerDakon** [comment]
ping?

**Pull request closed without merging**